### ISSUE TITLE
add `detectArrays` option to `no-magic-numbers` rule

### ISSUE THREAD

**Issue by mistadikay** [eslint/eslint#4188]
Sometimes we need to set an array of numbers, for example a list of breakpoints in a responsive layout. In this case it would be weird to put every number of this array into the variable. So maybe it makes sense to add a `detectArrays` option to `no-magic-numbers` rule?

```js
// in this case `no-magic-numbers` rule is being triggered
const responsiveLayoutBreakpoints = [ 460, 600, 800, 1000, 1200 ];
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
I don't think I agree that this exception makes sense. Those are magic numbers. There is no explanation as to what they are. A new developer who just started might not have any idea what they mean. That's what this rule is trying to prevent.

**Comment by mistadikay** [comment]
I don't quite follow why does it make sense for objects and doesn't for arrays. Could you please clarify the difference? As for the `no explanation` part, it seems to me that variable name `responsiveLayoutBreakpoints` is clear enough. Otherwise what are the alternatives? Put each breakpoint to its own variable?

**Comment by alberto** [comment]
Just disable the rule for that line. See the end of [this section](http://eslint.org/docs/user-guide/configuring#configuring-rules)

**Comment by ilyavolodin** [comment]
The point of this rule is to not have arbitrary numbers sprinkled throughout the codebase. If you need to use a number, you should declare a variable (constant, really) and use that instead. I think the same thing still applies to your example. I agree with @alberto just disable the rule either for this line or in general.

**Comment by nzakas** [comment]
@mistadikay the suggestion here is to manually disable the rule using either `/*eslint-disable*/` or `/*eslint-disable-line*/` for the one-off scenarios you have.

**Comment by mistadikay** [comment]
Okay. Thank you for your time.
